### ISSUE TITLE
imageUrl is undefined for users who have Gravatars

### ISSUE THREAD

**Issue by nscarlson** [FusionAuth/fusionauth-issues#619]
### Description
Inferring from this article here, it seems that FusionAuth should be returning an `imageUrl` for retrieved users' Gravatars without implementing any additional code: https://fusionauth.io/blog/2019/03/14/like-avatar-can-keep

> If you already have a Gravatar account then you don’t need to do anything else, we’ve taken care of everything. For those without a Gravatar account you’ll still see a randomly generated Gravatar. Everyone wins.

Though Gravatar images are displaying perfectly in the `Users` page and its list of users, the `imageUrl` field remains undefined (unless otherwise explicitly set) when retrieving users.  I'm currently using the `retrieveUserByEmail` function from the TS client, and haven't tested to see if the other methods are unaffected.

### Workaround
The Gravatar URLs can be generated with custom code, using the code snippet in the aforementioned article as a reference.

**Comment by mooreds** [comment]
Just so I understand, where would you like the `imageUrl` to be made accessible?

**Comment by robotdan** [comment]
This is working as designed. Not sure it makes sense for FusionAuth to return a generated image URL on the User API. The main purpose (from my point of view) of Gravatar for display purposes in the UI. 

You can build the same Image URL in our own app to display a Gravatar. Perhaps I could be convinced otherwise. 

The primary use case would so that you don't need to know anything about Gravatar in your own app and just rely upon the User object to return a URL that can used to display a user profile image of some sort?

**Comment by nscarlson** [comment]
>You can build the same Image URL in our own app to display a Gravatar. Perhaps I could be convinced otherwise.

That's what I'm doing currently.  Maybe I'm misreading the article, maybe it's only meant to refer to functionality inside of FusionAuth's admin ui.

**Comment by robotdan** [comment]
> That's what I'm doing currently. Maybe I'm misreading the article, maybe it's only meant to refer to functionality inside of FusionAuth's admin ui.

That is correct.  But perhaps there is an argument to be made that we could return this generated URL from an API, etc. 

We can discuss internally.

**Comment by DaviddH** [comment]
Would be great if this also was exposed in the API, that way we can re-use the profile picture logic from fusionauth and the profile pictures seen by admins/support staff and users at the front-end side are the same.

**Comment by robotdan** [comment]
The Gravatar logic is quite simple, they provide examples in many languages. If you want to implement this today, you can use their examples.  http://en.gravatar.com/site/implement/
